### ISSUE TITLE
Multiple gpus speed up training just a little

### ISSUE THREAD

**Issue by xqfone070** [open-mmlab/mmaction2#949]
I'm training SlowFast On AVA2.1. But I find multiple gpus speed up training just a little. How would I do to make the model training faster?

- Training with single gpu, it takes about 7days. 

  - command line: 
```python tools/train.py configs/detection/ava/slowfast_kinetics_pretrained_r50_4x16x1_20e_ava_rgb.py --validate```
   - logs:
```
2021-06-21 16:03:43,088 - mmaction - INFO - Epoch [1][1020/36876]	lr: 3.280e-03, eta: 7 days, 7:55:00, time: 0.853, data_time: 0.001
2021-06-21 16:04:00,166 - mmaction - INFO - Epoch [1][1040/36876]	lr: 3.283e-03, eta: 7 days, 7:53:19, time: 0.854, data_time: 0.001
2021-06-21 16:04:17,191 - mmaction - INFO - Epoch [1][1060/36876]	lr: 3.287e-03, eta: 7 days, 7:51:03, time: 0.851, data_time: 0.000
```

- Training with 4 gpus, it takes about 6 days.
  - command line: 
`./tools/dist_train.sh configs/detection/ava/slowfast_kinetics_pretrained_r50_4x16x1_20e_ava_rgb.py 4 --validate`
  - logs:
```
2021-06-21 15:25:02,016 - mmaction - INFO - Epoch [1][520/9219]	lr: 3.442e-03, eta: 5 days, 21:54:54, time: 2.912, data_time: 0.001
2021-06-21 15:25:53,463 - mmaction - INFO - Epoch [1][540/9219]	lr: 3.454e-03, eta: 5 days, 21:30:33, time: 2.572, data_time: 0.044
2021-06-21 15:26:40,063 - mmaction - INFO - Epoch [1][560/9219]	lr: 3.466e-03, eta: 5 days, 20:41:17, time: 2.329, data_time: 0.238
```

**Comment by irvingzhang0512** [comment]
Maybe the bottleneck is io or cpu, instead of gpu

**Comment by irvingzhang0512** [comment]
related issue #264

**Comment by xqfone070** [comment]
If the bottleneck is up or cpu,I think the data_time should be very big. In my training log,the data_time could be ignored.

**Comment by irvingzhang0512** [comment]
Not necessarily. The `data_time` shows the time cost of the **CURRENT** iter, not **AVERAGE** time. So it's possible that there are some iters that `date_time` are very large but not showed.

Better to follow [this comment](https://github.com/open-mmlab/mmaction2/issues/264#issuecomment-712560134) to check io/cpu/gpu utility first

**Comment by xqfone070** [comment]
Thank you, I will check it later!

**Comment by xqfone070** [comment]
@irvingzhang0512 Hi, I checked the cpu usage and disk usage. Disk(ssd) usage is below 15%, cpu usage is about 80%-90%, the sys usage of cpu is a little high, nearly 40%. Any suggestion for me?
![GPUx4_nmon_disk](https://user-images.githubusercontent.com/50127382/122851010-d469c000-d340-11eb-8534-ab8e3e16f979.png)
![GPUx4_htop_wa](https://user-images.githubusercontent.com/50127382/122851174-1a268880-d341-11eb-8263-8aa3764e3899.png)

**Comment by irvingzhang0512** [comment]
I'm not an expert for this. Don't know the cause.
I would recommend to reduce batch size.
BTW, I met similar issue once and solved it by update cuda and pytorch version(cuda92 to cuda110, pytorch1.3 to 1.6), not sure whether it helps or not.
@innerlee any suggestions?

**Comment by innerlee** [comment]
It seems that cpu is very busy. Here is a shot in the dark: put this line in the begining of `tools/train.py`:

```python
import cv2
cv2.setNumThreads(1)
```

**Comment by xqfone070** [comment]
@innerlee @irvingzhang0512 Yea, it works. Now, the cpu load average is about 10% and train total time is about 2.5days. Really thanks! 
Could you explain why it works?
![GPUx4_htop](https://user-images.githubusercontent.com/50127382/122863332-24eb1880-d355-11eb-8966-4cbfaae95c6c.png)
![GPUx4_train_log](https://user-images.githubusercontent.com/50127382/122863337-287e9f80-d355-11eb-92a2-39b79afd5d08.png)

**Comment by innerlee** [comment]
@dreamerlin We may put these two lines in our train.py? You may have a more thorough test

**Comment by xqfone070** [comment]
The reason of abnormal cpu usage  is a bug in OpenCV: OpenCV is unsafe to use with fork mode when OpenCV's multithreading is enabled. So we disable OpenCV's multithreading by `cv2.setNumThreads(1)`.
[https://github.com/pytorch/pytorch/issues/1838](url)
